### ISSUE TITLE
SBUS Output

### ISSUE THREAD

**Issue by robthomson** [iNavFlight/inav#5648]
Would it be possible to enable as PWM channel - or a serial channel to be configured to output sbus or ppm?

Sbus would be good. This would enable a simple solution to drive multiple sbus servos from the flight controller.

Functionally.. on bigger models this can simply things as you can then use remote sbus to pwm convertors out on the wings - net result.. less wiring!

**Comment by digitalentity** [comment]
Should be doable.

**Comment by robthomson** [comment]
Fantastic.  Would be really helpful.   I dare say easier for many than then i2c servo expander!

**Comment by digitalentity** [comment]
Actually, this is a very low hanging fruit. I'll look into it for 2.5

**Comment by robthomson** [comment]
Great stuff.  No major rush my end.. But will really simplify future builds.

Goodnto know its in the pipe line.

**Comment by digitalentity** [comment]
@robthomson I might have a draft ready by weekend, but I don't have any SBUS servos at hand. Do you have a setup you can test with?

**Comment by robthomson** [comment]
Yup.  I can do this easily.

Another simple option... 

Feed the sbus into another flight controller that can receive sbus. 

But more than happy to test this end if you want. want

**Comment by helicocrasher** [comment]
Fantastic, interrested in this function & could do some testing with discrete sbus decoders

**Comment by advancetom** [comment]
this would be also useful for use as a helicopter flybarless mode!!

**Comment by robthomson** [comment]
Hoping to test this again tomorrow.. been a bit snowed under past couple of days!

**Comment by Enggmaug** [comment]
Hello, I struggle finding how to know/define where this SBUS output is mapped to ?
I suppose it won't be the same as the SBUS input from the receiver... or will it ?

**Comment by advancetom** [comment]
Follow

**Comment by moyouyilansefengye** [comment]
I am a little confused about outputting sbus. I am running Inav 3.0.0 on my F4V3 and configurator is also 3.0.0.
I want to use an FC to convert crsf signal to sbus. Is that possible?

**Comment by VancouverUmbrella** [comment]
Pass-through SBUS output from a FC is just what I'm looking for. 

It's not clear to me whether the request for SBUS output in iNAV has been implemented or not.

Can anyone set me straight and point me to documentation?

Many thanks.

**Comment by Enggmaug** [comment]
I put the following :
```
serial 2 4194304 115200 115200 0 115200
set servo_protocol = SBUS
```

And it jsut worked outputing SBUS on Serial 3. 
Need then a Serial to SBUS inverter, and SBUS to Servos converter.

G.

**Comment by VancouverUmbrella** [comment]
Wonderful!

Thank you for that. Much appreciated.

Kindly,
Andrew

**Comment by froqstar** [comment]
@Enggmaug do you happen to know where to find documentation on that feature, e.g. how you came up with that CLI command? I struggle to find anything...

**Comment by Enggmaug** [comment]
> 

Hey, I don't quite remember, I believe it came from some forum discussions I found in google.
In the end, I'm not entirely satisfied, my setup wasn't very reliable. But at this point, I'm not sure whether it is due to the Inav implementation, the FC, the Sbus to PWM converter, or overall wiring and power distribution.
What happens is that sometimes, all servos take a random position and won't move anymore... it is possibly a power issue, since I use 7 servos on a biplan... 
I gave this up. I'd very much like to try out a similar solution using crossfire, though. I have a mateksys CSRF to PWM converter that allows for a better power distribution. I already made the feature request. Not sure it will be implemented, though...

G.
